docker/prepare-dev-environment/solr/docker-compose.yml

```diff
@@ -0,0 +1,113 @@
+version: "3.6"
+
+services:
+  solr-populator:
+    build: ${DOCKERFILE_PATH}
+    networks:
+      - atlas-test-net
+    depends_on:
+      - postgres
+      - solrcloud-0
+      - solrcloud-1
+    volumes:
+      - gradle-wrapper-dists:/root/.gradle/wrapper/dists
+      - gradle-ro-dep-cache:/gradle-ro-dep-cache:ro
+      - atlas-data-bioentity-properties:/atlas-data/bioentity_properties
+      - atlas-data-exp:/atlas-data/exp
+      - atlas-data-expdesign:/atlas-data/expdesign
+    environment:
+      JAVA_TOOL_OPTIONS: "-Dfile.encoding=UTF8"
+      EXP_IDS: ${EXP_IDS}
+      SPECIES: ${SPECIES}
+      GRADLE_RO_DEP_CACHE: /gradle-ro-dep-cache
+      POSTGRES_HOST: ${POSTGRES_HOST}
+      POSTGRES_DB: ${POSTGRES_DB}
+      POSTGRES_USER: ${POSTGRES_USER}
+      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
+      ZK_HOSTS: "${PROJECT_NAME}-${SOLR_CLOUD_ZK_CONTAINER_1_NAME}:2181,${PROJECT_NAME}-${SOLR_CLOUD_ZK_CONTAINER_2_NAME}:2181,${PROJECT_NAME}-${SOLR_CLOUD_ZK_CONTAINER_3_NAME}:2181"
+      SOLR_HOSTS: "http://${PROJECT_NAME}-${SOLR_CLOUD_CONTAINER_1_NAME}:8983/solr,http://${PROJECT_NAME}-${SOLR_CLOUD_CONTAINER_2_NAME}:8983/solr"
+      SOLR_HOST: ${SOLR_CLOUD_CONTAINER_1_NAME}:8983
+      SOLR_NUM_SHARDS: 2
+      NUM_DOCS_PER_BATCH: 20000
+      SOLR_COLLECTION_BIOENTITIES: bioentities
+      SOLR_COLLECTION_BIOENTITIES_SCHEMA_VERSION: 1
+      SOLR_COLLECTION_BULK_ANALYTICS: bulk-analytics
+      SOLR_COLLECTION_BULK_ANALYTICS_SCHEMA_VERSION: 1
+    working_dir: /root
+    command:
+      - |
+        for _SPECIES in $${SPECIES}
```

Comment:
Do we need double $ sign here?

Reply:
Yes, otherwise Docker Compose tries to interpolate the variables when the YAML file is evaluated, before starting the container. This caused many headaches.

Reply:
I don't get it why it is working with only a single $ sign in other places.

Reply:
Maybe you need it in bash commands but not when using it in variable definitions?

Reply:
I removed it in this case, but from today’s chat I think the use case for `$$` and `$` is clear now.